### PR TITLE
KEYCLOAK-18590 Realm localizations of one realm must not affect themes displayed in context of other realms

### PR DESCRIPTION
**Pull request by artur-baltabayev**

### CODE REVIEW

**Comment by hmlnarik** [comment]
Thank you for the PR! The code changes look good to me. Could you please add a test that would verify the behaviour?

**Comment by artur-baltabayev** [comment]
@hmlnarik Hello, thank you very much for reviewing the task so quickly and sorry for the late response! Meanwhile we were working on the test and settled with a single realm test case as a multi realm setup would be unnecessarily complex. A quick explanation: If the realm localization key is deleted, the login page should obviously no longer contain the value that was attached to the key. This wasn't the case earlier because the key and it's value were cached alongside the theme.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/i18n/LoginPageTest.java [hunk 1]
```diff
@@ -242,6 +248,31 @@ public void languageUserUpdates() {
...
 
+    // KEYCLOAK-18590
+    @Test
+    public void realmLocalizationMessagesAreNotCachedWithinTheTheme() {
+        final String locale = Locale.ENGLISH.toLanguageTag();
+
+        final String realmLocalizationMessageKey = "loginAccountTitle";
+        final String realmLocalizationMessageValue = "Localization Test";
+
+        CloseableHttpClient httpClient = (CloseableHttpClient) new HttpClientBuilder().build();
```
Comment (hmlnarik) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/i18n/LoginPageTest.java:259]: Ensire that you always close any `Response` and `HttpClient` objects, unclosed connections affect stability of the testsuite. Ideally use `try`-with-resources construct to account for the failure as well, see e.g. https://github.com/keycloak/keycloak/blob/c29b8c73a108125511693ec337bc6bb8e9e15cea/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/SAMLServletAdapterTest.java#L1365-L1373 for correct sample.
Also see `org.keycloak.testsuite.util.AdminClientUtil.createResteasyClient()` for preferred method to create RestEasy client

**Review by hmlnarik — CHANGES_REQUESTED** [review]
Thank you for the updates. The test idea looks good to me, only technical detail needs to be fixed. Feel free to fix the same in the `acceptLanguageHeader` test which seems to be used as a template for the newly added test.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/i18n/LoginPageTest.java (hunk 1 above)
Comment (artur-baltabayev) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/i18n/LoginPageTest.java]: Hello Hynek, thank you a lot again for the feedback :) .
I put all auto-closables into try-with-resources. The Resteasy client has to be closed manually at the end though.

**Review by hmlnarik — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**